Correções LiquidBackgroundService/RabbitMqConsumer

Dervanil opened the pull request:
Adição de async/await no metodo ProcessMessageAsync
Correções RabbitMqConsumer, exceção sem tratamentos direto, fazendo a worker reiniciar, adição de log para estes casos.

resolve [issue114](https://github.com/Avanade/Liquid-Application-Framework/issues/114)

bruno-brant commented:
I'm forcing this merge since it's only a CI update and right now reviewers are busy.

bruno-brant commented:
This closes #99.

Dervanil committed:
- Correções LiquidBackgroundService adição de async/await no metodo ProcessMessageAsync
- Correções Liquid Comsumer, remoção sem tratamento direto, que fazia worker reiniciar, adição de log para estes casos.

lucianareginalino commented on src/Liquid.Messaging.RabbitMq/RabbitMqConsumer.cs:
```diff
@@ -83,7 +86,7 @@ protected async Task MessageHandler(BasicDeliverEventArgs deliverEvent, Cancella
                     _channelModel.BasicNack(deliverEvent.DeliveryTag, false, true);
 
-                throw new MessagingConsumerException(ex);
+                _logger.LogError(ex, ex.Message);
```
o log das excessões já é gerado no interceptor, e portanto essa gera duplicidade.

lucianareginalino commented on test/Liquid.Messaging.RabbitMq.Tests/RabbitMqConsumerTest.cs:
```diff
@@ -77,7 +80,7 @@ public async Task MessageHandler_WhenProcessExecutionFail_ThrowException()
             var task = MessageHandler(message, new CancellationToken());
 
-            await Assert.ThrowsAsync<MessagingConsumerException>(() => task);
+            _logger.Received(1);
```
o log das excessões já é gerado no interceptor, e portanto essa gera duplicidade.

lucianareginalino reviewed (changes requested):
favor remover logs, pois podem gerar duplicidade.

Dervanil committed:
- Remoção do log
- Mudança de versão do componente

Dervanil committed:
Merge pull request #1 from Avanade/main

Branch update

lucianareginalino commented on src/Liquid.Messaging.RabbitMq/RabbitMqConsumer.cs:
```diff
@@ -78,12 +78,12 @@ protected async Task MessageHandler(BasicDeliverEventArgs deliverEvent, Cancella
                 }
             }
-            catch (Exception ex)
+            catch (Exception)
```
a exceção deve ser capturada para que seja propagada no contexto da execução.

lucianareginalino commented on src/Liquid.Messaging.RabbitMq/RabbitMqConsumer.cs:
```diff
@@ -78,12 +78,12 @@ protected async Task MessageHandler(BasicDeliverEventArgs deliverEvent, Cancella
+                {
                     _channelModel.BasicNack(deliverEvent.DeliveryTag, false, true);
-
-                throw new MessagingConsumerException(ex);
```
a exeção deve ser propagada para que o log seja registrado na telemetria.

lucianareginalino commented on test/Liquid.Messaging.RabbitMq.Tests/RabbitMqConsumerTest.cs:
```diff
@@ -61,26 +63,6 @@ public async Task MessageHandler_WhenProcessExecutedSucessfully()
             await MessageHandler(message, new CancellationToken());
         }
 
-        [Fact]
```
favor incluir o teste de falha, a exeção deve ser propagada.

lucianareginalino reviewed (changes requested):
por favor, realizar as correções descritas nos comentários.
